packages/cli/openapi-ir-to-fern/src/utils/convertSdkGroupName.ts

```diff
@@ -0,0 +1,26 @@
+import { FERN_PACKAGE_MARKER_FILENAME_NO_EXTENSION } from "@fern-api/configuration";
+import { RelativeFilePath } from "@fern-api/fs-utils";
+import { SdkGroupName } from "@fern-api/openapi-ir-sdk";
+import { camelCase } from "lodash-es";
+
+export function convertSdkGroupNameToFileWithoutExtension(groupName: SdkGroupName | undefined): string {
+    if (groupName == null || groupName.length === 0) {
+        return FERN_PACKAGE_MARKER_FILENAME_NO_EXTENSION;
+    }
+    const fileNames: string[] = [];
+    for (const [index, group] of groupName.entries()) {
+        if (typeof group === "string") {
+            fileNames.push(camelCase(group));
+        } else if (index < groupName.length - 1) {
+            fileNames.push(camelCase(group.name));
+        } else {
```

Comment:
can we check the type field here so that if additional types are added to that undiscriminated union we are forced to visit

Reply:
e.g. an assert never? Can do!